api/config/v1/flags.go

```diff
@@ -63,6 +63,7 @@ type CommandLineFlags struct {
 	MOFEDEnabled      *bool                   `json:"mofedEnabled"               yaml:"mofedEnabled"`
 	UseNodeFeatureAPI *bool                   `json:"useNodeFeatureAPI"          yaml:"useNodeFeatureAPI"`
 	Plugin            *PluginCommandLineFlags `json:"plugin,omitempty"           yaml:"plugin,omitempty"`
+	Mode              *string                 `json:"mode"                       yaml:"mode"`
```

Comment:
Nit: Let's add this above `Plugin` to so that the plugin and gfd-specific flags are last.